Magic wand didn't handle transitions well

bsb20 opened the pull request:
```
-When leaving bitmapSelect, the work area was not update until after
creating of the new tool
-Magic wand would then use an old copy of the work area
-Added line to update this copy of the work area when this
transition occurs
```

Commit:
Magic wand didn't handle transitions well

    -When leaving bitmapSelect, the work area was not update until after
    creating of the new tool
    -Magic wand would then use an old copy of the work area
    -Added line to update this copy of the work area when this
    transition occurs

cwillisf commented on src/svgeditor/BitmapEdit.as:
```diff
@@ -247,6 +247,7 @@ public class BitmapEdit extends ImageEdit {
 				// User was editing an object and switched tools, bake the object
 				bakeIntoBitmap();
 				saveToCostume();
+				targetCostume.segmentationState.unmarkedBitmap = workArea.getBitmap().bitmapData.clone();
```
As a future enhancement, do you think we could skip this most of the time and only do it when activating the magic wand, or something like that?

cwillisf commented:
:+1:

Commit:
Added magic wand refresh method

    -Must refresh when costume is updated while in magic wand tool

Commit:
Added "flip" handling when in magic wand

    -Creates a new segmentation state and flips it
    -Also, handled associated magic wand undo/redo bug

cwillisf commented:
These latest changes look good, but it looks like you accidentally pulled in an extra file (and an extra commit?) which modified `MediaLibrary.as`
